store/store.go

```diff
@@ -261,24 +264,66 @@ func (s *Store) GetModel(name string) *Model {
 
 // Reduce processes txn events into the models.
 func (s *Store) Reduce(events []core.Event) error {
-	codecActions, err := s.eventcodec.Reduce(events, s.datastore, baseKey)
+	txn, err := s.datastore.NewTransaction(false)
 	if err != nil {
 		return err
 	}
-	actions := make([]Action, len(codecActions))
-	for i, ca := range codecActions {
+	defer txn.Discard()
+
+	actions := make([]Action, len(events))
+	for i, e := range events {
+		key := baseKey.ChildString(e.Model()).ChildString(e.EntityID().String())
 		var actionType ActionType
```

Comment:
`store.Action` (as used by Listeners) has its own `ActionType` again, is this intentional, or could we also use core.ActionType here as well? It would simplify things a bit it we stuck to a single ActionType, and we actually wouldn't need to have the Type on as many structs/interfaces that way?